plugins/query-insights/src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.settings;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightsExporterType;
+
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Settings for Query Insights Plugin
+ *
+ * @opensearch.api
+ * @opensearch.experimental
+ */
+public class QueryInsightsSettings {
+    /**
+     * Default Values and Settings
+     */
+    public static final TimeValue MAX_WINDOW_SIZE = new TimeValue(21600, TimeUnit.SECONDS);
+    public static final int MAX_N_SIZE = 100;
+    public static final TimeValue MIN_EXPORT_INTERVAL = new TimeValue(21600, TimeUnit.SECONDS);
+    public static final String DEFAULT_LOCAL_INDEX_MAPPING = "mappings/top_n_queries_record.json";
+    /** Default window size in seconds to keep the top N queries with latency data in query insight store */
+    public static final int DEFAULT_WINDOW_SIZE = 60;
+    /** Default top N size to keep the data in query insight store */
+    public static final int DEFAULT_TOP_N_SIZE = 3;
```

Comment:
Are the window_size, top_N setting going to be same across different dimensions like latency, CPU and memory?

Reply:
Can discuss this when we design the cpu and memory aspects. To keep things consistent and not overcomplicate the config might make sense to have the same window size for all ?

Reply:
I don't think there's a valid point to separate the default values for different metrics types? But I'm also open to suggestions / opinions.